pkg/segment/test/segment_test.go

```diff
@@ -98,6 +110,62 @@ func TestAddRandom(t *testing.T) {
 			t.Errorf("Iteration %d: %v", i, err)
 			break
 		}
+		if err := s.checkMaxGap(); err != nil {
+			t.Errorf("When inserting %d: %v", j, err)
+			break
+		}
+	}
+	if got, want := countSegmentsIn(&s), nrInsertions; got != want {
+		t.Errorf("Wrong final number of segments: got %d, wanted %d", got, want)
+	}
+	if t.Failed() {
+		t.Logf("Insertion order: %v", order[:nrInsertions])
+		t.Logf("Set contents:\n%v", &s)
+	}
+}
+
+func TestAddRandomWithRandomInterval(t *testing.T) {
+	var s Set
+	order := randIntervalPermutation(testSize)
+	//rand.Seed(time.Now().UnixNano())
```

Comment:
I dropped the random here but I'm just wondering what is common practice about this? Having some random will possibly test more corner cases but make the test not reproductive.

Reply:
Purely random tests are discouraged, but it's sometimes useful (in concert with other tests for specific edge cases, of course). I think you shouldn't seed with the time, because you want them to be deterministic at least. The default behavior of the rand package starts with rand.Seed(1), which is fine.

Reply:
I agree that deterministic is important to reproduce any bug and error so I followed your suggestion to remove any time based seed. 

But as 8000 is not actually a very large number, with only one random combination of the test sequence, it's not likely to reveal all potential problems. Actually during development I spotted a bug through this.